backend/open_webui/routers/auths.py

```diff
@@ -669,30 +671,90 @@ async def signup(request: Request, response: Response, form_data: SignupForm):
+                        iss.lower(),
+                    ):
+                        openid_config_url += f"?appid={aud}"
+
+                async with ClientSession() as session:
+                    async with session.get(openid_config_url) as resp:
                         if resp.status == 200:
                             openid_data = await resp.json()
                             logout_url = openid_data.get("end_session_endpoint")
                             if logout_url:
                                 response.delete_cookie("oauth_id_token")
 
+                                # Construct post_logout_redirect_uri (absolute URL to signin page)
+                                post_logout_redirect_uri = None
+                                if (
+                                    WEBUI_AUTH_SIGNOUT_REDIRECT_URL
+                                    and WEBUI_AUTH_SIGNOUT_REDIRECT_URL.startswith(
+                                        "http"
+                                    )
+                                ):
+                                    post_logout_redirect_uri = (
+                                        WEBUI_AUTH_SIGNOUT_REDIRECT_URL
+                                    )
+                                elif request.app.state.config.WEBUI_URL:
+                                    post_logout_redirect_uri = (
+                                        f"{request.app.state.config.WEBUI_URL}/signin"
+                                    )
+                                else:
+                                    # Fallback to constructing from request
+                                    host = request.headers.get("host", "localhost")
+                                    scheme = request.scheme
+                                    post_logout_redirect_uri = (
+                                        f"{scheme}://{host}/signin"
+                                    )
+
+                                # Build full logout URL with params
+                                full_logout_url = (
+                                    f"{logout_url}?id_token_hint={oauth_id_token}"
+                                )
+                                if post_logout_redirect_uri:
+                                    full_logout_url += f"&post_logout_redirect_uri={quote(post_logout_redirect_uri)}"
+                                else:
+                                    log.warning(
+                                        "No post_logout_redirect_uri constructed; provider may not redirect back properly."
+                                    )
+
+                                if aud and re.search(
+                                    r"(microsoftonline\.com|microsoftonline\.us|partner\.microsoftonline\.cn|microsoftonline\.de)",
+                                    iss.lower(),
+                                ):
+                                    full_logout_url += f"&client_id={aud}"
```

Comment:
This `full_logout_url` should be added to the single oauth provider details in the `config.py` so that the URL handling is kept together.